Replaced annotation with ingressClassName (#30171)
kubernetes/website: content/en/docs/concepts/services-networking/ingress-controllers.md

## Using multiple Ingress controllers

> You may deploy any number of ingress controllers using [ingress class](https://kubernetes.io/docs/concepts/services-networking/ingress/#ingress-class)
> within a cluster. Note the `.metadata.name` of your ingress class resource. When you create an ingress you would need that name to specify the `ingressClassName` field on your Ingress object (refer to [IngressSpec v1 reference](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#ingressspec-v1-networking-k8s-io)). `ingressClassName` is a replacement of older [annotation method](https://kubernetes.io/docs/concepts/services-networking/ingress/#deprecated-annotation).

Reviewer: ```suggestion
You may deploy any number of ingress controllers using [ingress class](/docs/concepts/services-networking/ingress/#ingress-class)
within a cluster. Note the `.metadata.name` of your ingress class resource. When you create an ingress you would need that name to specify the `ingressClassName` field on your Ingress object (refer to [IngressSpec v1 reference](/docs/reference/generated/kubernetes-api/v1.22/#ingressspec-v1-networking-k8s-io)). `ingressClassName` is a replacement of older [annotation method](/docs/concepts/services-networking/ingress/#deprecated-annotation).
```

Reviewer 2: Even better: link to the _new_ API reference page: [`/docs/reference/kubernetes-api/service-resources/ingress-v1/#IngressSpec`](https://k8s.io/docs/reference/kubernetes-api/service-resources/ingress-v1/#IngressSpec).

I'd link to https://kubernetes.io/docs/reference/labels-annotations-taints/#kubernetes-io-ingress-class-deprecated for the deprecated annotation (we could add more text to that entry at the same time).

> If you do not define a class name, your cloud provider may use a default ingress controller. [Default IngressClass](https://kubernetes.io/docs/concepts/services-networking/ingress/#default-ingress-class) can be set using the `ingressclass.kubernetes.io/is-default-class` annotation `true` on an `IngressClass` resource.

Reviewer: ```suggestion
If you do not define a class name, your cloud provider may use a default ingress controller. [Default IngressClass](/docs/concepts/services-networking/ingress/#default-ingress-class) can be set using the `ingressclass.kubernetes.io/is-default-class` annotation `true` on an `IngressClass` resource.
```

Reviewer 2: ```suggestion
If you do not specify an IngressClass for an Ingress, and your cluster has exactly one IngressClass marked as default, then Kubernetes [applies](/docs/concepts/services-networking/ingress/#default-ingress-class) the cluster's default IngressClass to the Ingress.
You mark an IngressClass as default by setting the [`ingressclass.kubernetes.io/is-default-class` annotation](/docs/reference/labels-annotations-taints/#ingressclass-kubernetes-io-is-default-class) on that IngressClass, with the string value `"true"`.
```
